endpoints/events/vtrack_test.go

```diff
@@ -498,6 +498,92 @@ func TestShouldSendToCacheExpectedPutsAndUpdatableBiddersWhenBidderVastAllowed(t
 	assert.Contains(t, string(mockCacheClient.Values[1].Data), "bidder=updatable_bidder")
 }
 
+func TestShouldSendToCacheExpectedPutsAndUpdatableCaseSensitiveBiddersWhenBidderVastAllowed(t *testing.T) {
+	// mock pbs cache client
+	mockCacheClient := &vtrackMockCacheClient{
+		Fail:  false,
+		Uuids: []string{"uuid1", "uuid2"},
+	}
+
+	// mock AccountsFetcher
+	mockAccountsFetcher := &mockAccountsFetcher{
+		Fail: false,
+	}
+
+	// config
+	cfg := &config.Configuration{
+		MaxRequestSize: maxSize, VTrack: config.VTrack{
+			TimeoutMS: int64(2000), AllowUnknownBidder: false,
+		},
+		AccountDefaults: config.Account{},
+	}
+	cfg.MarshalAccountDefaults()
+
+	// bidder info
+	bidderInfos := make(config.BidderInfos)
+	bidderInfos["appnexus"] = config.BidderInfo{
+		Disabled:                false,
+		ModifyingVastXmlAllowed: true,
+	}
+
+	d, err := getVTrackRequestData(true, true)
+	assert.NoError(t, err)
+
+	cacheReq := &BidCacheRequest{
+		Puts: []prebid_cache_client.Cacheable{
+			{
+				Type:       prebid_cache_client.TypeXML,
+				BidID:      "bidId1",
+				Bidder:     "APPNEXUS", // case sensitive name
+				Data:       d,
+				TTLSeconds: 3600,
+				Timestamp:  1000,
+			},
+			{
+				Type:       prebid_cache_client.TypeXML,
+				BidID:      "bidId2",
+				Bidder:     "ApPnExUs", // case sensitive name
+				Data:       d,
+				TTLSeconds: 3600,
+				Timestamp:  1000,
+			},
+		},
+	}
+	buf := &bytes.Buffer{}
+	enc := json.NewEncoder(buf)
+	enc.SetEscapeHTML(false)
+	err = enc.Encode(cacheReq)
+	assert.NoError(t, err)
+	data := buf.String()
+
+	req := httptest.NewRequest("POST", "/vtrack?a=events_enabled", strings.NewReader(data))
+
+	recorder := httptest.NewRecorder()
+	e := vtrackEndpoint{
+		Cfg:                 cfg,
+		BidderInfos:         bidderInfos,
+		Cache:               mockCacheClient,
+		Accounts:            mockAccountsFetcher,
+		normalizeBidderName: openrtb_ext.NormalizeBidderName,
+	}
+
+	// execute
+	e.Handle(recorder, req, nil)
+
+	d, err = io.ReadAll(recorder.Result().Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// validate
+	assert.Equal(t, 200, recorder.Result().StatusCode, "Expected 200 when account is not found and request is valid")
+	assert.Equal(t, "{\"responses\":[{\"uuid\":\"uuid1\"},{\"uuid\":\"uuid2\"}]}", string(d), "Expected 200 when account is found and request is valid")
+	assert.Equal(t, "application/json", recorder.Header().Get("Content-Type"))
+	assert.Len(t, mockCacheClient.Values, 2)
+	assert.Contains(t, string(mockCacheClient.Values[0].Data), "bidder=APPNEXUS")
+	assert.Contains(t, string(mockCacheClient.Values[1].Data), "bidder=ApPnExUs")
```

Comment:
https://github.com/prebid/prebid-server/blob/11f8fa62a4e32716d2ffb47552485c99ecc543bc/endpoints/events/vtrack_test.go#L533-L549

assert to make sure adapter names from /vtrack endpoint are used to prepare vastXML string